packages/web-react/src/components/Toast/ToastBar.tsx

```diff
@@ -0,0 +1,76 @@
+import classNames from 'classnames';
+import React, { MutableRefObject, useRef } from 'react';
+import { Transition, TransitionStatus } from 'react-transition-group';
+import { useStyleProps } from '../../hooks';
+import { SpiritToastBarProps } from '../../types';
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { VisuallyHidden } from '../VisuallyHidden';
+import {
+  DEFAULT_TOAST_COLOR,
+  ICON_BOX_SIZE,
+  TOAST_BAR_CLOSE_BUTTON_LABEL_DEFAULT,
+  TRANSITIONING_STYLES,
+  TRANSITION_DURATION,
+} from './constants';
+import { useToastBarStyleProps } from './useToastBarStyleProps';
+import { useToastIcon } from './useToastIcon';
+
+const ToastBar = (props: SpiritToastBarProps) => {
+  const {
+    id,
+    children,
+    closeLabel = TOAST_BAR_CLOSE_BUTTON_LABEL_DEFAULT,
+    color = DEFAULT_TOAST_COLOR,
+    hasIcon,
+    iconName,
+    isDismissible,
+    isOpen = true,
+    onClose = () => {},
+    ...restProps
+  } = props;
+  const rootElementRef: MutableRefObject<HTMLDivElement | null> = useRef(null);
+  const toastIconName = useToastIcon({ color, iconName });
+  const { classProps, props: modifiedProps } = useToastBarStyleProps({
+    ...restProps,
+    color,
+    isDismissible,
+    id,
+  });
+  const { styleProps, props: otherProps } = useStyleProps(modifiedProps);
+
+  return (
+    <Transition in={isOpen} nodeRef={rootElementRef} timeout={TRANSITION_DURATION} unmountOnExit>
+      {(transitionState: TransitionStatus) => (
+        <div
+          {...styleProps}
+          {...otherProps}
+          className={classNames(classProps.root, TRANSITIONING_STYLES[transitionState], styleProps.className)}
+          ref={rootElementRef}
+        >
+          <div className={classProps.content}>
+            {(hasIcon || iconName) && <Icon name={toastIconName} boxSize={ICON_BOX_SIZE} />}
+            <div className={classProps.message}>{children}</div>
+          </div>
+
+          {isDismissible && onClose && (
+            <Button
```

Comment:
Just caught my eye: in a few components there is a `ComponentCloseButton`. Can we do it here the same way? The intention is to be able to refactor this to a common CloseButton component. Thanks.

Reply:
created new component for close button 👍